Match Briefcase export behavior when repeats are submitted to Google Sheets

grzesiek2010 opened the pull request:
Closes #2063 

#### What has been done to verify that this works as intended?
I've tested a form with `instanceID` and without (both with nested repeatable groups).

#### Why is this the best possible solution? Were any other approaches considered?
I added two columns to identify repeatable groups (`PARENT_KEY` and `KEY`) like in Briefcase.

#### Are there any risks to merging this code? If so, what are they?
I think it's not risky but a few different forms should be tested to be sure everything is ok.

#### Do we need any specific form for testing your changes? If so, please attach one.
I used:
[nRepeatsWithoutInstanceID.xml.txt](https://github.com/opendatakit/collect/files/1850844/nRepeatsWithoutInstanceID.xml.txt)
[nRepeats.xml.txt](https://github.com/opendatakit/collect/files/1850845/nRepeats.xml.txt)

grzesiek2010 committed:
Match Briefcase export behavior when repeats are submitted to Google Sheets

lognaturel commented on collect_app/src/main/java/org/odk/collect/android/tasks/InstanceGoogleSheetsUploader.java:
```diff
@@ -371,24 +379,33 @@ private void createSheetsIfNeeded(TreeElement element) throws UploadException {
+            answers.put(PARENT_KEY, parentKey);
+        }
+        if (hasRepeatableGroups(element)) {
+            answers.put(KEY, key);
```
The `KEY` is useful for reconstructing order even when there aren't further nested groups. Also, including `KEY` for all repeats matches Briefcase. So this should go in the prior `if`.

lognaturel commented on collect_app/src/main/java/org/odk/collect/android/tasks/InstanceGoogleSheetsUploader.java:
```diff
@@ -371,24 +379,33 @@ private void createSheetsIfNeeded(TreeElement element) throws UploadException {
+            columnTitles.add(PARENT_KEY);
+        }
+        if (hasRepeatableGroups(element)) {
+            columnTitles.add(KEY);
```
Same as above. Let's include the `KEY` for all repeats.

lognaturel committed:
Always include the KEY for repeats

Match Briefcase behavior and satisfy feature request from https://forum.opendatakit.org/t/repeat-groups-and-google-sheets-upload/6677/20

lognaturel commented:
I know @grzesiek2010 is asleep so I made the quick change and this is ready for a final review and hopefully merge.

mmarciniak90 commented:
Tested with success as a part of https://github.com/opendatakit/collect/pull/2066

Verified on Android: 4.1, 4.4, 6.0, 7.0, 8.0

Verified cases: 
* nested-indexed-repeat.xml
* nRepeatsithoutInstanceID.xml
* nRepeats.xml
* fewRepeatGroupsGoogleSheets.xml

@opendatakit-bot label "behavior verified"
